python/lsst/meas/algorithms/testUtils.py

```diff
@@ -84,3 +84,47 @@ def plantSources(bbox, kwid, sky, coordList, addPoissonNoise=True):
     exposure.setPsf(psf)
 
     return exposure
+
+
+def makeRandomTransmissionCurve(rng, minWavelength=4000.0, maxWavelength=7000.0, nWavelengths=200,
+                                maxRadius=80.0, nRadii=30, perturb=0.05):
+    """Create a random TransmissionCurve with nontrivial spatial and
+    wavelength variation.
+
+    Parameters
+    ----------
+    rng : numpy.random.RandomState
```

Comment:
`RandomState` is a class with different methods for different distributions; it can always do both discrete and continuous, depending on the method called.

Reply:
yes, but then what is the meaning of subtracting off the 0.5, as it seems to be just shifting the central tendency of the distribution. I though you had it to somehow "center" the integer draws of the discrete pdf

Reply:
I'm not actually using a discrete distribution; `rand()` returns a uniform random number between `0` and `1`; this transforms it into a random number between `-0.5` and `0.5`.  I them multiply that by `2` to get a distribution between `-1` and `1`, and then scale by the given pertubation amount.